### PR TITLE
✨ Enable `Small File Downloads` (#2091)

### PR DESCRIPTION
**Pull request by mistryrn**
**Description of changes**

* Add File download button back to File Entity page
* Connect to new `/get-download-url` endpoint of File Storage API

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- UI Kit (select one):
  - [ ] New Components with storybook stories created
  - [ ] Modified Existing components and updates stories
  - [x] No new UIKit components or changes
- [x] Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

**Comment by demariadaniel** [comment]
tested locally, disabled states are working fine
tooltip may need to be updated for users w/o daco-approval
api is not merged so download is not working yet

**Review by demariadaniel — CHANGES_REQUESTED** [review]
tested locally, disabled states are working fine
tooltip may need to be updated for users w/o daco-approval
api is not merged so download is not working yet

File: components/pages/file-repository/utils/constants.tsx [hunk 1]
```diff
@@ -83,3 +83,5 @@ export const fileRepoTableTSVColumns: FileRepositoryTSVColumn[] = [
     getter: 'object_id',
   },
 ];
+
+export const MAX_FILE_DOWNLOAD_SIZE = 100000000; // 100MB
```
Comment (joneubank) [inline components/pages/file-repository/utils/constants.tsx:87]: Should we make this a configurable value?

Comment (mistryrn) [reply components/pages/file-repository/utils/constants.tsx:87]: Good idea, I'll update it here and in the API to be configurable via env var

File: global/utils/stringUtils.ts [hunk 2]
```diff
@@ -25,3 +25,14 @@ export const capitalize = (s: string) =>
...
     .map((str) => str.replace(str[0], (str[0] || '').toUpperCase()))
     .join(' ');
+
+/**
+ * Extracts the filename from the content-disposition header
+ * @param contentDisposition - content-disposition header
+ * @returns {string} filename
+ */
+export const getFilename = (contentDisposition: string): string => {
+  const filenameRegex = /(?<=filename(?:=|\*=(?:[\w\-]+'')))["']?([^"';\n]+)["']?/gi;
```
Comment (mistryrn) [inline global/utils/stringUtils.ts:35]: Regex demo here: https://regex101.com/r/FkLhkr/1

Grabs whatever follows the `filename=` in the `Content-Disposition: attachment; filename="filename.txt"` header, handles a few special cases that might occur.

Comment (ciaranschutte) [reply global/utils/stringUtils.ts:35]: sick! that's for the demo

**Review by demariadaniel — APPROVED** [review]

**Pull request merged**